eks.go

```diff
@@ -0,0 +1,106 @@
+/******************************************************************************
+Cloud Resource Counter
+File: eks.go
+
+Summary: Provides a count of all EKS nodes.
+******************************************************************************/
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+	color "github.com/logrusorgru/aurora"
+)
+
+// EKSNodes retrieves the count of all EKS Nodes either for all
+// regions (allRegions is true) or the region associated with the
+// session. This method gives status back to the user via the supplied
+// ActivityMonitor instance.
+func EKSNodes(sf ServiceFactory, am ActivityMonitor, allRegions bool) int {
+	nodeCount := 0
+
+	errs := make([]string, 0)
+
+	// Indicate activity
+	am.StartAction("Retrieving EKS Node counts")
+
+	// Create a new instance of the EKS service
+	if allRegions {
+		// Get the list of all enabled regions for this account
+		regionsSlice := GetEC2Regions(sf.GetEC2InstanceService(""), am)
+
+		// Loop through all of the regions
+		for _, regionName := range regionsSlice {
+			nodeCount += eksCountForSingleRegion(regionName, sf, am, &errs)
+		}
+	} else {
+		// Get the EC2 counts for the region selected by this session
+		nodeCount += eksCountForSingleRegion("", sf, am, &errs)
+	}
+
+	// Indicate end of activity
+	am.EndAction("OK (%d)", color.Bold(nodeCount))
+
+	// Print list of errors that happened while retrieving node counts
+	for _, err := range errs {
+		am.SubResourceError(err)
+	}
+
+	return nodeCount
+}
+
+func eksCountForSingleRegion(region string, sf ServiceFactory, am ActivityMonitor, errs *[]string) int {
+	// Indicate activity
+	am.Message(".")
+
+	// Retrieve an EKS service
+	eksSvc := sf.GetEKSService(region)
+
+	// Construct our input to find all Clusters
+	input := &eks.ListClustersInput{}
+
+	nodeCount := 0
+	err := eksSvc.ListClusters(input, func(clusterList *eks.ListClustersOutput, lastPage bool) bool {
```

Comment:
**Suggestion _(non-blocking)_:**
Since `lastPage` isn't used, you can explicitly not set it to anything which can be helpful so readers don't go looking for its use.
```suggestion
	err := eksSvc.ListClusters(input, func(clusterList *eks.ListClustersOutput, _ bool) bool {
```